internal/controller/apiextensions/claim/configurator.go

```diff
@@ -126,13 +126,19 @@ func (c *APIDryRunCompositeConfigurator) Configure(ctx context.Context, cm resou
 	ucp.SetClaimReference(proposed)
 
 	if !meta.WasCreated(cp) {
-		// The API server returns an available name derived from
-		// generateName when we perform a dry-run create. This name is
-		// likely (but not guaranteed) to be available when we create
-		// the composite resource. If the API server generates a name
-		// that is unavailable it will return a 500 ServerTimeout error.
-		cp.SetGenerateName(fmt.Sprintf("%s-", cm.GetName()))
-		return errors.Wrap(c.client.Create(ctx, cp, client.DryRunAll), errName)
+		// If the claim already specifies a resource ref, use the existing name.
+		// Otherwise let the API server generate a new one.
+		if ref := cm.GetResourceReference(); ref != nil && ref.Name != "" {
+			cp.SetName(ref.Name)
+		} else {
```

Comment:
Nit: I have a weird(?) bias against using `else` - you could also just `return nil` after line 132 and keep this "non-exceptional" case a little less nested.